### PR TITLE
TSA-4 Commerz Bank API

### PR DESCRIPTION
**Pull request by abteilung6**
Added

- Commerz Bank API
- Standardized Authorization Code Grant Flow

### CODE REVIEW

File: bank_connector/banks/deutsche_bank.py [hunk 1]
```diff
@@ -11,24 +11,17 @@ class Mode(Enum):
...
-    def get_authorize_url(self, client_id: str, redirect_uri: str):
-        """Get the url for login and grant access pages"""
-        return self._format_authorize_url(client_id, redirect_uri)
-
-    def _format_authorize_url(self, client_id: str, redirect_uri: str) -> str:
+    def get_authorization_url(self, client_id: str, redirect_uri: str) -> str:
         path = "/gw/oidc/authorize"
-        query = "?" + urlencode(dict(response_type="code", client_id=client_id, redirect_uri=redirect_uri))
-        return urljoin(self.base_url, path + query)
+        return AuthorizationCodeGrant.format_authorization_url(self.base_url, path, client_id, redirect_uri)
```
Comment (steven-mi) [inline bank_connector/banks/deutsche_bank.py]: I think this should be `self`?

File: bank_connector/banks/commerz_bank.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+
+class CommerzBankClient(BankClientBase, AuthorizationCodeGrant):
+
+    def __init__(self):
+        super().__init__(Mode.Simulation.value)
+
+    def get_authorization_url(self, client_id: str, redirect_uri: str) -> str:
+        path = "/auth/realms/sandbox/protocol/openid-connect/auth"
+        return AuthorizationCodeGrant.format_authorization_url(self.base_url, path, client_id, redirect_uri)
```
Comment (steven-mi) [inline bank_connector/banks/commerz_bank.py]: I think this should be `self`?

**Review by steven-mi — APPROVED** [review]
LGTM

File: bank_connector/banks/deutsche_bank.py (hunk 1 above)
Comment (abteilung6) [reply bank_connector/banks/deutsche_bank.py]: true

File: bank_connector/banks/commerz_bank.py (hunk 2 above)
Comment (abteilung6) [reply bank_connector/banks/commerz_bank.py]: true

**Pull request merged**